### DISCUSSION TITLE
Is there a way to call shopware 6 layout on checkout pages?

### DISCUSSION THREAD

**Discussion by ghost** [shopware/frontends#381] [Q&A] [answered]
So I have a requirement where the Cart, Checkout process, and Success pages have some CMS blocks in design. However, I haven't found a way to assign a layout from the Shopware 6 admin to these pages. Is it possible to achieve this? if yes! i will highly appreciate guidance in this matter. 🙏

**Comment by BrocksiNet** [comment, marked as answer]
Hi, @junaidfar00qui, and thx for this question. 

It is true that you can not assign a layout to these pages (Cart/Checkout). But you can always load any CMS page via `store API`. You must create the page via Admin first and after that use the hard-coded UUID inside the [Request](https://shopware.stoplight.io/docs/store-api/70c7b54c9faf9-fetch-and-resolve-a-cms-page). After that, you can use the data from JSON to create whatever you need or pass the data to existing components. The main issue here is that the UUID can be different depending on instances and it feels haky to have hardcoded UUID.

**More complex way:**
You can search for the CMS page with some `admin API` request, like `POST /api/search/cms-page` for example. To do this you could use the new [API client](https://www.npmjs.com/package/@shopware/api-client/v/0.0.0-canary-20230822131551) (see section about Admin Setup). And then we maybe need to think about when will this API call happen ... maybe prerender would be nice to only do this during build time once. This will lead to thinking about building again when something is changed in Admin. Or you add some cache e.g. 1 day so most of the time the content is served from the cache. Have a look at the NUXT docu [here](https://nuxt.com/docs/guide/concepts/rendering#hybrid-rendering).

I hope this reply helps you. :handshake:
